regparse.c

```diff
@@ -736,10 +747,51 @@ onig_names_free(regex_t* reg)
   return 0;
 }
 
+extern int
+onig_names_copy(regex_t* reg, regex_t* oreg)
+{
+  NameTable* ot = oreg->name_table;
+  if (ot) {
+    OnigEncoding enc = oreg->enc;
+    int i, num = ot->num;
+    NameTable* t = xmalloc(sizeof(*t));
+    CHECK_NULL_RETURN_MEMERR(t);
+    *t = *ot;
+    t->e = xmalloc(t->alloc * sizeof(t->e[0]));
+    if (IS_NULL(t->e)) {
+      xfree(t);
+      return ONIGERR_MEMORY;
+    }
+    t->num = 0;
+    reg->name_table = t;
+    for (i = 0; i < num; t->num = ++i) {
+      NameEntry* oe = &(ot->e[i]);
+      NameEntry* e = &(t->e[i]);
+      *e = *oe;
+      e->name = NULL;
+      e->back_refs = NULL;
+      e->name = strdup_with_null(enc, oe->name, oe->name + e->name_len);
+      if (IS_NULL(e->name)) {
+        onig_names_free(reg);
+        return ONIGERR_MEMORY;
+      }
+      e->back_refs = xmalloc(e->back_alloc * sizeof(e->back_refs[0]));
+      if (IS_NULL(e->back_refs)) {
+        xfree(e->name);
+        onig_names_free(reg);
+        return ONIGERR_MEMORY;
+      }
+      memcpy(e->back_refs, oe->back_refs, e->back_num * sizeof(e->back_refs[0]));
```

Comment:
I think we need to manually set `back_ref1` here. I believe it is always the same as `back_refs[0]`

```suggestion
      memcpy(e->back_refs, oe->back_refs, e->back_num * sizeof(e->back_refs[0]));
      e->back_ref1 = e->back_refs[0];
```